tests/editing/uat-testing/uat_testing.py

```diff
@@ -0,0 +1,509 @@
+        # The resulting indices are stored in the mismatch_locations variable.
+        # The tilde (~) is a unary operator in Python that performs a bitwise negation.
+        # In this case, it is used to negate the comparison variable.
+        # This means that the condition inside np.where() will be true for elements where comparison is false.
+        mismatch_locations = np.where(~comparison)
+        file_failures = []  # Store the failures for each file
+
+        # This for loop iterates over each pair of row and column indices in the mismatch_locations variable.
+        # The mismatch_locations variable contains the indices where the comparison between the processed output
+        # and expected output dataframes is false.
+        # These indices represent the locations where the values in the two dataframes do not match.
+        for row, col in zip(*mismatch_locations):
+            # Check the method type to determine the failure details format
+            if method == "TCC":
+                # Create a failure dictionary with the reference, row number, column number, expected output, and processed output
+                failure = {
+                    "reference": df_processed_output.loc[row, "reference"],
+                    "row": row + 1,
+                    "col": col + 1,
+                    "expected_output": df_expected_output.iloc[row, col],
+                    "processed_output": df_processed_output.iloc[row, col],
+                }
+            elif method == "TPC":
+                # Create a failure dictionary with the RU, row number, column number, expected output, and processed output
+                failure = {
+                    "RU": df_processed_output.loc[row, "RU"],
+                    "row": row + 1,
+                    "col": col + 1,
+                    "expected_output": df_expected_output.iloc[row, col],
+                    "processed_output": df_processed_output.iloc[row, col],
+                }
+            # Append the failure dictionary to the file_failures list
+            file_failures.append(failure)
+
+        # If there are any failures in the file, create a dictionary with the file name and the list of failures
+        if len(file_failures) > 0:
+            file_failure = {"file_name": file1, "failures": file_failures}
+            return file_failure
+
+
+# This test_values function is used to test the values in the processed output CSV files against the expected output CSV files.
+# It uses the parametrize decorator to run the test with different arguments.
+# The test_data parameter is a tuple containing the method ("TCC" or "TPC"), the path to the original test data directory,
+# and the path to the processed test data directory.
+@pytest.mark.parametrize(
+    "test_data",
+    [
+        ("TCC", "tcc_test_data_original/", "tcc_test_data_processed/"),
+        ("TPC", "tpc_test_data_original/", "tpc_test_data_processed/"),
+    ],
+)
+def test_values(test_data):
```

Comment:
compare_values may have been a more descriptive name and is consistent with your compare_dataframes function

Reply:
Renamed!

Reply:
The way works is pytest automatically discovers and runs tests based on certain naming conventions.

For a function to be recognised as a test and run by pytest, it must be a function whose name starts with test_. So when you name your function test_values, pytest recognises it as a test and runs it.

When you rename the function to compare_values, pytest no longer recognises it as a test (because it doesn't start with test_), so it doesn't run it.

But we can use `pytest -k compare_values` command to by pass this if we were to run a test under the name compare_values.